packages/imperative/src/imperative/src/plugins/cmd/install/install.handler.ts

```diff
@@ -71,112 +74,218 @@ export default class InstallHandler implements ICommandHandler {
-                            const pluginName = await install(packageArgument, packageInfo.registry, true);
-                            params.response.console.log("Installed plugin name = '" + pluginName + "'");
-                            params.response.console.log(runValidatePlugin(pluginName));
+                            params.response.console.log(
+                                "Plug-ins within the Imperative CLI Framework can legitimately gain\n" +
+                                    `control of the ${ImperativeConfig.instance.rootCommandName} CLI application ` +
+                                    "during the execution of every command.\n" +
+                                    "Install 3rd party plug-ins at your own risk.\n"
+                            );
+                            params.response.console.log(
+                                "Location = " + installRegistry
+                            );
+
+                            params.response.console.log(
+                                "\n_______________________________________________________________"
+                            );
+                            const pluginName = await install(
+                                packageArgument,
+                                packageInfo.location,
+                                true
+                            );
+                            params.response.console.log(
+                                "Installed plugin name = '" + pluginName + "'"
+                            );
+                            params.response.console.log(
+                                runValidatePlugin(pluginName)
+                            );
                         }
                     }
+                }
+
+                for (const plugin of params.arguments.plugin ?? []) {
+                    // Get the registry to install to
+                    if (typeof params.arguments.registry === "undefined") {
+
+                        let isDirTest: boolean;
+
+                        try{
+                            isDirTest = IO.isDir(plugin);
+                        }
+                        catch(e){
+                            isDirTest = false;
+                        }
 
-                    // write the json file when done if not the plugin json file
-                } else {
-                    for (const packageString of params.arguments.plugin) {
-                        params.response.console.log("\n_______________________________________________________________");
-                        const pluginName = await install(`${packageString}`, installRegistry);
-                        params.response.console.log("Installed plugin name = '" + pluginName + "'");
-                        params.response.console.log(runValidatePlugin(pluginName));
+                        if (plugin.startsWith("@")) {
+                            installRegistry = getScopeRegistry(
```

Comment:
It seems like we are calling `getScopedRegistry` twice for scoped plug-ins, first in the loop above to add the location to `plugins.json`, and then again here. Since `getScopedRegistry` runs an npm cli command, it may take a long time to run in some environments. Would it make sense to cache the results of `getScopedRegistry` from earlier and reuse them here?

Reply:
resolved